src/fields/elements/getItemElementFields.ts

```diff
@@ -1,59 +1,92 @@
+      .then(entries => new Map(entries));
+
+type TermData = Readonly<{
+  id: string;
+  name: string;
+  codename: string;
+}>;
+
+const createTermData = (term: TaxonomyModels.Taxonomy): ReadonlyArray<TermData> => [
+  {
+    id: term.id,
+    name: term.name,
+    codename: term.codename,
+  },
+  ...term.terms.flatMap(createTermData),
+];
 
 export type ElementFields = Readonly<{
   [key: `elements__${string}`]: string | string[] | number | undefined;
 }>;
 
-function getSimpleElementField(element: ContentTypeElements.ContentTypeElementModel) {
-  switch (element.type) {
-    case 'text':
-    case 'rich_text':
-    case 'custom':
-      return getField(element, { type: 'text' });
-
-    case 'number':
-      return getField(element, { type: 'float' });
-
-    case 'date_time':
-      return getField(element, { type: 'datetime' });
-
-    case 'modular_content':
-      const opts = {
-        search: 'find_item.id',
-        list: true,
-        dynamic: 'get_linked_items.id.name',
-        type: 'string',
-        altersDynamicFields: false,
-      } as const;
-
-      return getField(element, opts);
-
-    case 'multiple_choice': {
-      const choices = element.options.map(o => ({ label: o.name, value: o.codename || '', sample: '' }));
-
-      return getField(element, { type: 'unicode', list: element.mode === 'multiple', choices });
+const getSimpleElementField = (termsByElementId: ReadonlyMap<string, readonly TermData[]>) =>
+  (element: ContentTypeElements.ContentTypeElementModel) => {
+    switch (element.type) {
+      case 'text':
+      case 'rich_text':
+      case 'custom':
+        return getField(element, { type: 'text' });
+
+      case 'number':
+        return getField(element, { type: 'float' });
+
+      case 'date_time':
+        return getField(element, { type: 'datetime' });
+
+      case 'modular_content':
+        const opts = {
+          search: 'find_item.id',
+          list: true,
+          dynamic: 'get_linked_items.id.name',
+          type: 'string',
+          altersDynamicFields: false,
+        } as const;
+
+        return getField(element, opts);
+
+      case 'multiple_choice': {
+        const choices = element.options.map(o => ({ label: o.name, value: o.codename || '', sample: '' }));
+
+        return getField(element, { type: 'unicode', list: element.mode === 'multiple', choices });
+      }
+      case 'asset':
+        return getField(element, { type: 'unicode', list: true });
+
+      case 'taxonomy': {
+        const terms = termsByElementId.get(element.id) ?? [];
+        const idChoices = terms.map(t => ({ value: t.id, label: t.name, sample: t.id }));
+        const codenameChoices = terms.map(t => ({ value: t.codename, label: `${t.name} (codename)`, sample: t.codename }));
+
+        return getField(element, { type: 'unicode', choices: [...idChoices, ...codenameChoices], list: true });
```

Comment:
Is it not possible to have two tabs one for id and one for codename?

Reply:
I am no sure what you mean by tabs, but it is possible to have two separate fields and then choose one of them at runtime. It is an unfortunate situation, because we need both options to be able to accept both identifiers from previous steps, but in the dropdown in the UI, we would ideally want just one of them.